### PR TITLE
Task av/j272 add forgot password functionality

### PR DESCRIPTION
**Pull request by artem-totality**

### CODE REVIEW

**Review by EugeniusGR — DISMISSED** [review]

**Review by JuliaDzyuba — DISMISSED** [review]

File: backend/src/services/auth/auth.service.ts [hunk 1]
```diff
@@ -70,6 +75,34 @@ class Auth {
...
+    const hasUser = Boolean(user);
+
+    if (!hasUser) {
+      throw new HttpError({
+        status: HttpCode.NOT_FOUND,
+        message: ErrorMessage.EMAIL_DOES_NOT_EXIST,
+      });
+    }
+
+    const newPassword = getRandomId(6);
```
Comment (what1s1ove) [inline backend/src/services/auth/auth.service.ts]: add const for it

File: backend/src/services/mailer/mailer.service.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  constructor({ mailerApiKey, mailerEmail }: Constructor) {
+    sgMail.setApiKey(mailerApiKey);
+    this.#mailerEmail = mailerEmail;
+  }
+
+  public async sendMail(message: MailTemplate): Promise<boolean> {
+    try {
+      await sgMail.send({
+        ...message,
+        from: this.#mailerEmail,
```
Comment (what1s1ove) [inline backend/src/services/mailer/mailer.service.ts]: let's pass it from params

File: backend/src/services/mailer/mailer.service.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  mailerApiKey: string;
+  mailerEmail: string;
+};
+
+class Mailer {
+  #mailerEmail: string;
+
+  constructor({ mailerApiKey, mailerEmail }: Constructor) {
+    sgMail.setApiKey(mailerApiKey);
+    this.#mailerEmail = mailerEmail;
```
Comment (what1s1ove) [inline backend/src/services/mailer/mailer.service.ts]:
```suggestion
```

File: shared/src/common/types/user/user-reset-password-payload.type.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { UserPayload } from './user-payload.type';
+
+type UserResetPasswordPayload = UserPayload;
```
Comment (what1s1ove) [inline shared/src/common/types/user/user-reset-password-payload.type.ts]:
```suggestion
type UserResetPasswordPayload = {
  // describe it
};
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**